amm/contracts/stable_pool/lib.rs

```diff
@@ -287,7 +295,7 @@ pub mod stable_pool {
             let current_time = self.env().block_timestamp();
             let mut rate_changed = false;
             for rate in self.pool.token_rates.iter_mut() {
-                rate_changed = rate_changed || rate.update_rate(current_time);
+                rate_changed = rate_changed | rate.update_rate(current_time);
```

Comment:
You could have left the `||` but change the order of elements so that `rate.update_rate(...)` is always called first.